docs/ROUTING_GUIDE.md

```
@@ -0,0 +1,223 @@
+```
+
+## Navigation
+
+### Using AppNavigator (Recommended)
+
+The `AppNavigator` class provides type-safe navigation methods:
+
+```dart
+import 'package:nashik/core/router/navigation/app_navigator.dart';
+
+// Navigate to a route (replaces current)
```

markdownlint-cli2 (0.20.0)

[warning] Trailing spaces
Expected: 0 or 2; Actual: 1

(MD009, no-trailing-spaces)